### DISCUSSION TITLE
Quickfix workflow

### DISCUSSION THREAD

**Discussion by fredrikaverpil** [folke/trouble.nvim#290] [Q&A] [answered]
Coming from vscode, I'm used to having a "Quick fix" action. Since this doesn't work the same way with Neovim, I was wondering if there's anything like this or what people's workflows are like in Neovim.

For example, in vscode, if I have a linter installed that complains about imports being unsorted, I can hit "Quick fix" and the imports will become sorted.

![image](https://github.com/folke/trouble.nvim/assets/994357/22e3cc5e-f4de-4798-b905-e5ece2f2bd96)

I could for example create a keymap that runs (or manually run) `:ruff --fix %` to fix the current file from this particular problem.

Any other suggestions?

**Comment by folke** [comment]
`<leader>ca` (code actions)

**Reply by fredrikaverpil** [reply]
Do you mean to say I should add my own code actions under `<leader>ca ...` in my `config/keymaps.lua`?
Are there reference docs on this or similar somewhere?

**Reply by fredrikaverpil** [reply]
I've added this to my `config/keymaps.lua`:

```lua
local keymap = vim.keymap

keymap.set("n", "<leader>car", ":!ruff --fix %<CR>")
```

But I'm getting this error when I start nvim now:

```
Invalid spec module: `plugins.code_actions`
Expected a `table` of specs, but a `boolean` was returned instead
```

**Reply by fredrikaverpil** [reply]
I feel I'm likely completely doing the wrong thing here. Thing is that if I run `:lua vim.lsp.buf.code_action()` there is no code action for ruff, and maybe it's because ruff is linting via null-ls.

**Reply by fredrikaverpil** [reply, marked as answer]
Okay, never mind, I had to install ruff-lsp with Mason. Now I'm all good and can get to the code actions via `<leader>ca`. Thanks @folke and sorry for the noise.
